mattermost/DEBIAN/postinst

```diff
@@ -1,7 +1,36 @@
 #!/usr/bin/env bash
 
+set -e
+IS_UPGRADE=false
```

Comment:
nit: you can do something like

```
TRUE_REG='^([tT][rR][uU][eE]|[yY]|[yY][eE][sS]|1)$'
FALSE_REG='^([fF][aA][lL][sS][eE]|[nN]|[nN][oO]|0)$'
```

and then do:
```
if [[ $IS_UPGRADE =~ $TRUE_REG ]]; then
```

Also is a good practice to add some debug flag and strict scripting to help us debug when is needed:
```
IS_DEBUG=${IS_DEBUG:-false}
if [[ $IS_DEBUG =~ $TRUE_REG ]]; then
    set -o xtrace
fi

IS_STRICT=${IS_STRICT:-false}
if [[ $IS_STRICT =~ $TRUE_REG ]]; then
    set -o errexit
    set -o nounset
    set -o pipefail
fi
```

Reply:
These are getting shipped with the mattermost debian packages. Are we sure we want to introduce debugging flags to our released packages? 

Example 
https://github.com/grafana/grafana/blob/main/packaging/deb/control/postinst

Reply:
Yeah I got it but I believe this will be really helpful if a customer mentions it doesn't work and somehow be able to debug and get some details back in which line etc. failed. In an on-prem environment will be super hard to debug. Thoughts?

Reply:
Valid point. 👍 
Will push a Commit to support debugging

Reply:
Done with: 


```bash
> sudo IS_DEBUG=true dpkg -i mattermost_7.3.1-0.deb
Selecting previously unselected package mattermost.
(Reading database ... 62439 files and directories currently installed.)
Preparing to unpack mattermost_7.3.1-0.deb ...
+ IS_STRICT=false
+ [[ false =~ ^([tT][rR][uU][eE]|[yY]|[yY][eE][sS]|1)$ ]]
+ case "$1" in
Unpacking mattermost (7.3.1-0) ...
Setting up mattermost (7.3.1-0) ...
+ IS_STRICT=false
+ [[ false =~ ^([tT][rR][uU][eE]|[yY]|[yY][eE][sS]|1)$ ]]
+ case "$1" in
+ '[' -z '' ']'
+ MATTERMOST_USER=mattermost
+ '[' -z '' ']'
+ MATTERMOST_GROUP=mattermost
+ getent group mattermost
+ addgroup --system mattermost --quiet
+ id mattermost
+ adduser --system --home /opt/mattermost --no-create-home --ingroup mattermost --disabled-password --shell /bin/false mattermost
+ chown -R mattermost:mattermost /opt/mattermost
+ '[' '' '!=' '' ']'
+ [[ ! false =~ ^([tT][rR][uU][eE]|[yY]|[yY][eE][sS]|1)$ ]]
+ systemctl daemon-reload
+ systemctl enable mattermost
> sudo IS_DEBUG=true dpkg -i mattermost_7.4.0-0.deb
(Reading database ... 66288 files and directories currently installed.)
Preparing to unpack mattermost_7.4.0-0.deb ...
+ IS_STRICT=false
+ [[ false =~ ^([tT][rR][uU][eE]|[yY]|[yY][eE][sS]|1)$ ]]
+ case "$1" in
+ IS_STRICT=false
+ [[ false =~ ^([tT][rR][uU][eE]|[yY]|[yY][eE][sS]|1)$ ]]
+ case "$1" in
+ systemctl stop mattermost
+ '[' 7.3.1-0 '!=' '' ']'
+ IS_UPGRADE=true
+ [[ true =~ ^([tT][rR][uU][eE]|[yY]|[yY][eE][sS]|1)$ ]]
+ find /opt/mattermost/ /opt/mattermost/client/ -mindepth 1 -maxdepth 1 '!' '(' -type d '(' -path /opt/mattermost/client -o -path /opt/mattermost/client/plugins -o -path /opt/mattermost/config -o -path /opt/mattermost/logs -o -path /opt/mattermost/plugins -o -path /opt/mattermost/data ')' -prune ')' -exec rm -r '{}' ';'
Unpacking mattermost (7.4.0-0) over (7.3.1-0) ...
+ IS_STRICT=false
+ [[ false =~ ^([tT][rR][uU][eE]|[yY]|[yY][eE][sS]|1)$ ]]
+ case "$1" in
Setting up mattermost (7.4.0-0) ...
+ IS_STRICT=false
+ [[ false =~ ^([tT][rR][uU][eE]|[yY]|[yY][eE][sS]|1)$ ]]
+ case "$1" in
+ '[' -z '' ']'
+ MATTERMOST_USER=mattermost
+ '[' -z '' ']'
+ MATTERMOST_GROUP=mattermost
+ getent group mattermost
+ id mattermost
+ chown -R mattermost:mattermost /opt/mattermost
+ '[' 7.3.1-0 '!=' '' ']'
+ IS_UPGRADE=true
+ [[ ! true =~ ^([tT][rR][uU][eE]|[yY]|[yY][eE][sS]|1)$ ]]
+ systemctl daemon-reload
+ systemctl start mattermost
```

Works great 🚀